ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/AverageProfitCriterion.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 Marc de Verdelhan, 2017-2021 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.analysis.criteria;
+
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.Position;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.num.Num;
+import org.ta4j.core.utils.NumUtils;
+
+/**
+ * This is simple average profit of all winning positions
+ */
+public class AverageProfitCriterion extends AbstractAnalysisCriterion {
+
+    private boolean isWinningPosition(BarSeries series, Position position) {
```

Comment:
Use `NumberOfWinningPositionsCriterion` instead of duplicating code.

Reply:
Not sure I understand what you mean. I checked the '_NumberOfWinningPositionsCriterion_' and it counts the number of positions that are profitable. The '_AverageProfitCriterion_' is counting **Average** of the profitable positions: 
**SUM of all profits / number of profitable positions**.
It does it in one go in the stream processing. 
If I included the other criterion, and used it to calculate the count, it would have been slower (the count is processed extra and then the average). 
But we can make a micro benchmark to check... although I strongly believe, my code would perform better than adding criterion call:) But I could be also mistaken... 

I also find the stream processing more readable... but this is matter of taste.

Reply:
Yes, I included Position#isWinning() and Position#isLoosing() in https://github.com/ta4j/ta4j/pull/704 which can be used here..